driver-core/src/main/com/mongodb/internal/connection/DescriptionHelper.java

```diff
@@ -77,14 +77,22 @@ static ConnectionDescription createConnectionDescription(final ClusterConnection
             connectionDescription = connectionDescription.withConnectionId(newConnectionId);
         }
         if (clusterConnectionMode == ClusterConnectionMode.LOAD_BALANCED) {
-            if (manufactureServiceId) {
-                TopologyVersion topologyVersion = getTopologyVersion(isMasterResult);
-                if (topologyVersion != null) {
-                    connectionDescription = connectionDescription.withServiceId(topologyVersion.getProcessId());
-                }
+            ObjectId serviceId = getServiceId(isMasterResult);
+            if (serviceId != null) {
+                connectionDescription = connectionDescription.withServiceId(serviceId);
             } else {
-                throw new MongoClientException("Driver attempted to initialize in load balancing mode, but the server does not support "
-                        + "this mode");
+                if (manufactureServiceId) {
```

Comment:
Nit: would be cleaner as `} else if (manufactureServiceId) {`

Reply:
Yeah, definitely.  There's even an IntelliJ auto-fix for this.

While looking at the code, I also realized I could simplify the manufactureServiceId branch by using an assertion instead of a conditional.  This is appropriate as it's test-only code.  PTAL.

Done.